client/service/src/task_manager/mod.rs

```diff
@@ -63,10 +83,13 @@ impl SpawnTaskHandle {
 	pub fn spawn(
 		&self,
 		name: &'static str,
-		group: Option<&'static str>,
+		group: impl Into<GroupName>,
 		task: impl Future<Output = ()> + Send + 'static,
 	) {
-		self.spawn_inner(name, group, task, TaskType::Async)
+		match group.into() {
+			GroupName::Specific(var) => self.spawn_inner(name, Some(var), task, TaskType::Async),
+			_ => (),
```

Comment:
this is not correct, we want to spawn in both cases
spawn_inner should probably also take `GroupName` instead of an `Option`

Reply:
OK. Will do.

Reply:
I have updated the spawner_inner and other methods too 
can you take a look at the PR again
@ordian